src/RuleFilterParser/QueryProvider/ApiQueryProvider.cs

```diff
@@ -134,13 +154,16 @@ public TResult Execute<TResult>(Expression expression)
         return (TResult)result;
```

Comment:
There should probably be a null check here, and return default(TResult) if the result is null perhaps?